neo/io/nwbio.py

```diff
@@ -0,0 +1,823 @@
+                        **self.annotations)  # todo: timeseries.control / control_description
+
+
+class EventProxy(BaseEventProxy):
+
+    def __init__(self, timeseries, nwb_group):
+        self._timeseries = timeseries
+        self.name = timeseries.name
+        self.annotations = {"nwb_group": nwb_group}
+        self.description = try_json_field(timeseries.description)
+        if isinstance(self.description, dict):
+            self.annotations.update(self.description)
+            self.description = None
+        self.shape = self._timeseries.data.shape
+
+    def load(self, time_slice=None, strict_slicing=True):
+        """
+        *Args*:
+            :time_slice: None or tuple of the time slice expressed with quantities.
+                            None is the entire signal.
+            :strict_slicing: True by default.
+                Control if an error is raised or not when one of the time_slice members
+                (t_start or t_stop) is outside the real time range of the segment.
+        """
+        if time_slice:
+            raise NotImplementedError("todo")
+        else:
+            times = self._timeseries.timestamps[:]
+            labels = self._timeseries.data[:]
+        return Event(times * pq.s,
+                     labels=labels,
+                     name=self.name,
+                     description=self.description,
+                     **self.annotations)
+
+
+class EpochProxy(BaseEpochProxy):
+
+    def __init__(self, epochs_table, epoch_name=None, index=None):
+        self._epochs_table = epochs_table
+        if index is not None:
+            self._index = index
+            self.shape = (index.sum(),)
+        else:
+            self._index = slice(None)
+            self.shape = epochs_table.n_rows  # untested, just guessed that n_rows exists
+        self.name = epoch_name
+
+    def load(self, time_slice=None, strict_slicing=True):
+        """
+        *Args*:
+            :time_slice: None or tuple of the time slice expressed with quantities.
+                            None is all of the intervals.
+            :strict_slicing: True by default.
+                Control if an error is raised or not when one of the time_slice members
+                (t_start or t_stop) is outside the real time range of the segment.
+        """
+        start_times = self._epochs_table.start_time[self._index]
+        stop_times = self._epochs_table.stop_time[self._index]
+        durations = stop_times - start_times
+        labels = self._epochs_table.tags[self._index]
+
+        return Epoch(times=start_times * pq.s,
+                     durations=durations * pq.s,
+                     labels=labels,
+                     name=self.name)
+
+
+class SpikeTrainProxy(BaseSpikeTrainProxy):
+
+    def __init__(self,  units_table, id):
+        self._units_table = units_table
+        self.id = id
+        self.units = pq.s
+        t_start, t_stop = units_table.get_unit_obs_intervals(id)[0]
```

Comment:
If I understand correctly, this will only work if there is a single observation interval. It will not work if there are no observation intervals, and if there are multiple, it will only output the first.

Reply:
added to #1013